website_and_docs/content/documentation/webdriver/bidi_apis.en.md

```diff
@@ -5,19 +5,56 @@ weight: 11
 aliases: ["/documentation/en/webdriver/bidi_apis/"]
 ---
 
-In Selenium 4, new Evented APIs were introduced that
-allow users to be able to capture events from the
-browser as they happen rather than using the
-traditional approach of Request/Response that
-WebDriver has used for other APIs.
+Selenium is working with Browser Vendors to create the 
+[BiDirectional WebDriver Protocol](https://w3c.github.io/webdriver-bidi/) 
+as a means to provide a stable, cross-browser implementation of a subset of the functionality found
+[Chrome DevTools](../../support_packages/chrome_devtools).
+The traditional webdriver model of strict request/response commands will be supplemented with the ability to
+stream events from the user agent to the controlling software via WebSockets, better matching the evented nature of the browser DOM.
 
-Internally WebDriver will create a WebSocket connection
-to the browser for events and commands to be transmitted.
+While this initiative is being developed, Selenium has implemented this API which is currently supported by
```

Comment:
I don't understand this sentence at all.

Reply:
Trying to set expectations for what is working now. how about:

> The BiDi API code in Selenium 4 currently works on Chrome and Edge, with some support for Firefox.

Reply:
There is no Bidi implementation yet. We're all CDP at the moment. Firefox is starting to ship the necessary pieces of the nascent spec, though.

How about:

> Because it's a bad idea to tie your tests to a specific version of a specific browser, the Selenium project recommends using WebDriver Bidi wherever possible. However, until the spec is complete there are many useful things that the CDP offers. To help keep your tests independent and portable, Selenium offers some useful helper classes. At the moment, these use the CDP, but when we shall be using WebDriver Bidi as soon as possible!